src/main/java/org/broadinstitute/hellbender/tools/walkers/genotyper/AlleleSubsettingUtils.java

```diff
@@ -112,19 +136,35 @@ public static GenotypesContext subsetAlleles(final GenotypesContext originalGs,
 
             // restrict AD to the new allele subset
             if(g.hasAD()) {
-                final int[] oldAD = g.getAD();
-                final int[] newAD = IntStream.range(0, allelesToKeep.size()).map(n -> oldAD[allelePermutation.fromIndex(n)]).toArray();
-                final int nonRefIndex = allelesToKeep.indexOf(Allele.NON_REF_ALLELE);
-                if (nonRefIndex != -1 && nonRefIndex < newAD.length) {
-                    newAD[nonRefIndex] = 0;  //we will "lose" coverage here, but otherwise merging NON_REF AD counts with other alleles "creates" reads
-                }
+                final int[] newAD = getNewAlleleBasedReadCountAnnotation(allelesToKeep, allelePermutation, g.getAD());
                 gb.AD(newAD);
+                // if we have recalculated AD and the original genotype had AF but was then removed, then recalculate AF based on AD counts
+                if (alleleBasedLengthAnnots.contains(GATKVCFConstants.ALLELE_FRACTION_KEY) && g.hasExtendedAttribute(GATKVCFConstants.ALLELE_FRACTION_KEY)) {
+                    final double[] newAFs = MathUtils.normalizeSumToOne(Arrays.stream(newAD).mapToDouble(x -> x).toArray());
+                    gb.attribute(GATKVCFConstants.ALLELE_FRACTION_KEY, Arrays.copyOfRange(newAFs, 1, newAFs.length)); //omit the first entry of the array corresponding to the reference
+                }
             }
             newGTs.add(gb.make());
         }
         return newGTs;
     }
 
+    /**
+     * Subsets allele length annotation to match the alleles that have been kept
+     * @param allelesToKeep     list of alleles that are not being filtered out
+     * @param allelePermutation permutation that matches the index from the old annotation to the new annotation given which alleles have been removed
+     * @param oldAnnot          the original annotation
+     * @return                  the new subset annotation
+     */
+    private static int[] getNewAlleleBasedReadCountAnnotation(List<Allele> allelesToKeep, Permutation<Allele> allelePermutation, int[] oldAnnot) {
+        final int[] newAnnot = IntStream.range(0, allelesToKeep.size()).map(n -> oldAnnot[allelePermutation.fromIndex(n)]).toArray();
+        final int nonRefIndex = allelesToKeep.indexOf(Allele.NON_REF_ALLELE);
+        if (nonRefIndex != -1 && nonRefIndex < newAnnot.length) {
+            newAnnot[nonRefIndex] = 0;  //we will "lose" coverage here, but otherwise merging NON_REF AD counts with other alleles "creates" reads
```

Comment:
To be honest, I can't follow my own logic here, but you just copied so you're not to blame.